de_web_plugin.cpp

```diff
@@ -4475,6 +4475,12 @@ void DeRestPluginPrivate::updateSensorNode(const deCONZ::NodeEvent &event)
 
                                 updateSensorEtag(&*i);
                             }
+                            else if (ia->id() == 0xff01) // Xiaomi magic
+                            {
+                                const char *str = ia->toString().toLatin1().constData();
```

Comment:
Careful with that one `.constData()` is only guaranteed to be valid as long as the object is on the stack or heap, caused me a lot of headaches in the past :)

One way to do it:

    QByteArray arr = ia->toString().toLatin1();
    const char *str = arr.constData();

Or easier:

    qPrintable(ia->toString().toLatin1().toHex(' '))

Reply:
The QT version on my Raspberry Stretch doesn't support an argument to `toHex()`, but,  Indeed, it's easier.  However it doesn't solve the real issue, see following annotated log:
```
Feb 19 22:17:27 pi deCONZ[9921]: 22:17:26:542 	payload: 050042156c756d692e73656e736f725f776c65616b2e61713101ff42220121d10b03281e0421a8330521540006240000000000082104020a211e67641000
                                                         --------l u m i . s e n s o r _ w l e a k . a q 1 --------====XX============XX========
Feb 19 22:17:27 pi deCONZ[9921]: 22:17:26:543 Node data 0x00158d0001bb6159 profileId: 0x0104, clusterId: 0x0000
Feb 19 22:17:27 pi deCONZ[9921]: 22:17:26:543 Update Sensor 0x00158D0001BB6159 Basic Cluster
Feb 19 22:17:27 pi deCONZ[9921]: 22:17:26:543 >>>>> 0x00158D0001BB6159: Xiaomi magic: 01213f0b03281e04213f33052154
                                                                                      ====XX============XX========
```
The payload of the _Report Attributes_ shows two attributes for cluster 0x0000: _Model Identifier_ (0x0005) and 0xff01, both of type string (0x42).  The length for 0xff01 (0x22) is one byte more than actually transmitted.  
In `updateSensorNode()`, the attribute payload is corrupted (see XX) and trimmed.  
The battery voltage is 0x0bd1, or 3.025V.